### ISSUE TITLE
missing samples during export

### ISSUE THREAD

**Issue by farvardin** [LMMS/lmms#3702]
Hello,

I'm using LMMS 1.2.0-rc2.151 which I've compiled myself (linux mint 17.3, with KDE 4.14)

I have an issue for exporting music with samples in it: sometimes some samples are missing, other times ALL the samples are missing. The music from LMMS instruments or VST or carla baypatch are always exported correctly.

First I've noticed Jack server was started but pulseaudio is bridged and LMMs is using the Alsa renderer.

I've stopped jack, exported again, and got all my samples mixed in the final export. But after a few new tried, I noticed even with jack stopped, sometimes some samples were missing as well, so I think it's not related to jack.

The odd thing is even when the export was defective, if I choose to export "all the tracks", then all the samples are correctly exported as single tracks.

**Comment by farvardin** [comment]
about some questions from another thread:

On my initial song I had some effects such as LMMS EQ or ladspa reverb on the samples (in flac format), but on my latest test it was only the raw samples, without any effects (converted from mp3 to wav).

I don't know what the "scope" is. The export is very fast because the test is tiny, I don't really see anything in the mixer vumeter. The samples are really simples. And the problem seems to occur at random.

**Comment by trollworkout** [comment]
There's an rc4 now for linux as an appimage. Look on @tresf 's  github he seems to be maintaining appimages now.

**Comment by Umcaruje** [comment]
>This isn't RC4, but fixes leading to its release.

Please note that, these are not official production RC4's but tests for the appimage. The actual RC4 will have an appimage added to it when it's released.

Also I fail to understand what an appimage would do to help this issue, since the OP already compiles himself and seems pretty up to date.

**Comment by trollworkout** [comment]
@Umcaruje  is not the appimage is the newer release which may or may have not have fixed that issue. the advice was to download the latest available release and try again see if that issue has been fixed or not. that's all.

**Comment by farvardin** [comment]
well I don't know what to say now. I've rebooted my computer, downloaded the appimage, and the export was correct. I've started again my self compiled 1.2.0-rc2.151 version and the export was correct as well on my previously problematic tracks (without jack). I've also started jack, and the export was correct both with the appimage version and my version.

So I assume part of the problem remains in my soundcard (C-Media Electronics, Inc. CM106 Like Sound Device), but I suppose if I got this problem once other people might encounter this as well. Was is bugging me is when it was in a state to get those random errors during export, it seemed to be always ok when exporting to multiple tracks.

I'll test further, and with another sound card.

**Comment by tresf** [comment]
So you have jack working with AppImage?   https://github.com/LMMS/lmms/pull/3688#issuecomment-315634489

**Comment by farvardin** [comment]
yes, through the jack bridge. I haven't tested on my other card yet, but I've recorded a project in a shared folder, I'll try asap (for this one the export is blank again on my main computer)

**Comment by farvardin** [comment]
I've tested again on another computer (this time with an integrated cheap sound module Audio device: Advanced Micro Devices, Inc. [AMD/ATI] Tobago HDMI Audio [Radeon R7 360 / R9 360 OEM]).

Jack is enabled, but again, with a bridge so I use LMMS with the Alsa backend. 

With the rc2 I've built myself, first time I've exported to wav, the track was empty. 
With the AppImage rc3, the exported track was ok.
With my rc2, second time I export => empty track
With my rc2, 3rd time I export => empty track
With my rc2, 4th time I export (now with 48000 instead of 44100 and 32 bit instead of 24 bit) => track is Ok
With my rc2, 5th time I export (back with 44100 and 24 bit) => track is Ok
With the AppImage, 2nd export is OK as well.
Again with AppImage, a 3rd export was ok.  But every time I export from the AppImage, the encoding counter get stuck at 99%, and I have to kill the LMMS processus because it is frozen.

So I'd say given the limited test I've made, the export from rc2 is not reliable (sometimes it works, other time it doesn't), and on rc3 it seems always correct. I'll try to recompile a new version with rc3 (I need Carla Baypatch so I just can't use the AppImage)

Maybe it might worth opening a new issue with the stucked counter in rc3, I'll test this further as well.
